include/ginkgo/core/base/dim.hpp

```diff
@@ -200,11 +215,154 @@ struct dim<1u, DimensionType> {
         return dim(x.first_ * y.first_);
     }
 
+    friend GKO_ATTRIBUTES std::ostream &operator<<(std::ostream &os,
+                                                   const dim &x)
+    {
+        os << x.first_;
+        return os;
+    }
+
 private:
     dimension_type first_;
 };
 
 
+/**
+ * A type representing the dimensions of a multidimensional batch object.
+ *
+ * @tparam Dimensionality  number of dimensions of the object
+ * @tparam DimensionType  datatype used to represent each dimension
+ *
+ * @ingroup batch_dim
+ */
+template <size_type Dimensionality = 2, typename DimensionType = size_type>
+struct batch_dim {
+    static constexpr size_type dimensionality = Dimensionality;
+    using dimension_type = DimensionType;
+
+    /**
+     * Checks if the batch_dim object stores equal sizes.
+     *
+     * @return bool representing whether equal sizes are being stored
+     */
+    bool stores_equal_sizes() const { return equal_sizes_; }
+
+    /**
+     * Get the number of batches stored
+     *
+     * @return num_batches
```

Comment:
```suggestion
     * @return The size of the batch.
```

Reply:
I dont think size here is very clear, because size can also mean the dimension of the entries or of the entire batch object. Maybe number of batch entries is better ?